yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -391,6 +391,50 @@ func (d *JsonQueryFunctionDetector) GetIssues() []QueryIssue {
 	return issues
 }
 
+type UniqueNullsNotDistinctDetector struct {
+	query    string
+	detected bool
+}
+
+func NewUniqueNullsNotDistinctDetector(query string) *UniqueNullsNotDistinctDetector {
+	return &UniqueNullsNotDistinctDetector{
+		query: query,
+	}
+}
+
+// Detect checks if a unique constraint is defined which has nulls not distinct
+func (d *UniqueNullsNotDistinctDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_INDEX_STMT_NODE {
+		indexStmt, err := queryparser.ProtoAsIndexStmt(msg)
+		if err != nil {
+			return err
+		}
+
+		if indexStmt.Unique && indexStmt.NullsNotDistinct {
+			d.detected = true
+		}
+	} else if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_TABLECONSTRAINT_NODE {
+		constraintNode, err := queryparser.ProtoAsTableConstraint(msg)
+		if err != nil {
+			return err
+		}
+
+		if constraintNode.Contype == queryparser.UNIQUE_CONSTR_TYPE && constraintNode.NullsNotDistinct {
+			d.detected = true
+		}
```

Comment:
not sure about this. but same might apply here.

Reply:
I had earlier implemented it using the protomsg approach. I had to use a new function to convert the enum number of Contype to the name of that contype like `UNIQUE_CONSTR_TYPE` to make it more readable and not susceptible to any changes in the enum numbers map. 
But the current approach seemed more direct and easier to read.
What do you think should be done here?